docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js

```diff
@@ -1,34 +1,44 @@
+import { Component } from "//unpkg.com/can@5/core.mjs";
+
 // 60 = 2π
 const base60ToRadians = (base60Number) =>
   2 * Math.PI * base60Number / 60;
 
-can.Component.extend({
+Component.extend({
   tag: "analog-clock",
-  view: '<canvas id="analog" width="255" height="255"></canvas>',
+  view: `<canvas this:to="canvasElement" id="analog" width="255" height="255"></canvas>`,
   ViewModel: {
+    // the canvas element
+    canvasElement: HTMLCanvasElement,
+
+    // the canvas 2d context
+    get canvas() {
+      return this.canvasElement.getContext("2d");
+    },
+
     connectedCallback(element) {
-      const canvas = element.firstChild.getContext("2d");
       const diameter = 255;
       const radius = diameter/2 - 5;
       const center = diameter/2;
+      
       // draw circle
-      canvas.lineWidth = 4.0;
-      canvas.strokeStyle = "#567";
-      canvas.beginPath();
-      canvas.arc(center, center, radius, 0, Math.PI * 2, true);
-      canvas.closePath();
-      canvas.stroke();
-
+      this.canvas.lineWidth = 4.0;
+      this.canvas.strokeStyle = "#567";
+      this.canvas.beginPath();
+      this.canvas.arc(center, center, radius, 0, Math.PI * 2, true);
+      this.canvas.closePath();
+      this.canvas.stroke();
+      
       this.listenTo("time", (ev, time) => {
-        canvas.clearRect(0, 0, diameter, diameter);
+        this.canvas.clearRect(0, 0, diameter, diameter);
 
         // draw circle
-        canvas.lineWidth = 4.0;
-        canvas.strokeStyle = "#567";
-        canvas.beginPath();
-        canvas.arc(center, center, radius, 0, Math.PI * 2, true);
-        canvas.closePath();
-        canvas.stroke();
+        this.canvas.lineWidth = 4.0;
+        this.canvas.strokeStyle = "#567";
+        this.canvas.beginPath();
+        this.canvas.arc(center, center, radius, 0, Math.PI * 2, true);
+        this.canvas.closePath();
+        this.canvas.stroke();
 
         Object.assign(canvas, {
```

Comment:
This should be `Object.assign(this.canvas, {`

Going through the guide from the starter CodePen and copying the highlighted code in each step should reveal any issues like this.

Reply:
Fixed!